ambry-messageformat/src/test/java/com/github/ambry/messageformat/MessageFormatRecordTest.java

```diff
@@ -748,9 +748,31 @@ private BlobData getBlobRecordV2(int blobSize, BlobType blobType, ByteBuffer blo
     return MessageFormatRecord.deserializeBlob(new ByteBufferInputStream(outputBuffer));
   }
 
+  /**
+   * Serializes the blob content using BlobRecord Verison 2 with the passsed in params
+   * De-serialized the blob returns the {@link BlobData} for the same
+   * @param blobSize
+   * @param blobType
+   * @param blobContent
+   * @param outputBuffer
+   * @return
+   * @throws IOException
+   * @throws MessageFormatException
+   */
+  private BlobData getBlobRecordV3(int blobSize, BlobType blobType, ByteBuffer blobContent, ByteBuffer outputBuffer)
+      throws IOException, MessageFormatException {
+    MessageFormatRecord.Blob_Format_V3.serializePartialBlobRecord(outputBuffer, blobSize, blobType, false);
+    outputBuffer.put(blobContent);
+    Crc32 crc = new Crc32();
+    crc.update(outputBuffer.array(), 0, outputBuffer.position());
+    outputBuffer.putLong(crc.getValue());
+    outputBuffer.flip();
+    return MessageFormatRecord.deserializeBlob(new ByteBufferInputStream(outputBuffer));
+  }
+
   @Test
   public void testBlobRecordWithMetadataContentV2() throws IOException, MessageFormatException {
-    // Test Blob V2 with actual metadata blob V2
+    // Test Blob V3 with actual metadata blob V2
```

Comment:
In my opinion, we should remove testing V1 and V2.  Just stay with V3.  The additional logic written to serialize using V1 will never happen.  We will never go back to start serializing V1 because it will be lost of data.  So let's not keep code that don't make sense to keep.  Instead, we should clean it up to remove dangerous code like serializing in older versions.